pkg/turntaker/turntaker_test.go

```diff
@@ -0,0 +1,160 @@
+package turntaker
+
+import (
+	"os"
+	"testing"
+
+	"github.com/google/uuid"
+
+	"github.com/openshift/node-observability-agent/pkg/runs"
+)
+
+const (
+	validUID string = "dd37122b-daaf-4d75-9250-c0747e9c5c47"
+)
+
+func TestTakeTurn(t *testing.T) {
+	testCases := []struct {
+		name           string
+		previousState  State
+		expectedState  State
+		errFileContent string
+		expectedError  bool
+	}{
+		{
+			name:          "turnTaker was free, can take turn",
+			previousState: Free,
+			expectedState: Free,
+			expectedError: false,
+		},
+		{
+			name:          "turnTaker was taken, returns the id of running job",
+			previousState: Taken,
+			expectedState: Taken,
+			expectedError: false,
+		},
+		{
+			name:           "turnTaker was in error, returns the id of job in error",
+			previousState:  InError,
+			expectedState:  InError,
+			errFileContent: "{\"ID\":\"" + validUID + "\",\"ProfilingRuns\":[{\"Type\":\"Kubelet\",\"Successful\":false,\"BeginDate\":\"2022-03-03T10:10:17.188097819Z\",\"EndDate\":\"2022-03-03T10:10:47.211572681Z\",\"Error\":\"fake error\"},{\"Type\":\"CRIO\",\"Successful\":true,\"BeginDate\":\"2022-03-03T10:10:17.188499431Z\",\"EndDate\":\"2022-03-03T10:10:47.215840909Z\",\"Error\":null}]}",
+			expectedError:  false,
+		},
+		{
+			name:           "turnTaker was in error, and unable to read error file, error",
+			previousState:  InError,
+			expectedState:  InError,
+			errFileContent: "{\"ID" + validUID + "\",\"ProfilingRuns\":{\"Type\":\"Kubelet\",\"Successful\":false,\"BeginDate\":\"2022-03-03T10:10:17.188097819Z\",\"EndDate\":\"2022-03-03T10:10:47.211572681Z\",\"Error\":\"fake error\"},{\"Type\":\"CRIO\",\"Successful\":true,\"BeginDate\":\"2022-03-03T10:10:17.188499431Z\",\"EndDate\":\"2022-03-03T10:10:47.215840909Z\",\"Error\":null}]}",
+			expectedError:  true,
+		},
+	}
+	for _, tC := range testCases {
+		t.Run(tC.name, func(t *testing.T) {
+			var previousJobId uuid.UUID
+			turnTaker := NewTurnTaker("/tmp/agent.err")
```

Comment:
Define const

Reply:
I modified this a bit because when running tests concurrently, the content of /tmp/agent.err was becoming inconsistent, thus rendering the results inconsistent as well.
I used a different error file name for each test case instead.